RestaurantsNearMe/Data/CloudKitService.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  CloudKitService.swift
+//  RestaurantsNearMe
+//
+//  Created by Colin Evans on 2023-11-27.
+//
+
+import Foundation
+import Combine
+import CloudKit
+
+class CloudKitService: CloudKitServiceProviding {
+  let applicationContainer = CKContainer.default()
+  private let _fetchingStatus = PassthroughSubject<Bool, Never>()
+
+  private let _accountStatus = CurrentValueSubject<CKAccountStatus, Never>(.couldNotDetermine)
+  var accountStatus: AnyPublisher<CKAccountStatus, Never> {
+    _accountStatus.eraseToAnyPublisher()
+  }
+  
+  init() {
+    Task.init {
```

Comment:
I'm always wary of doing things like this in an `init` - I'd rather have explicit control over when this is executed - also, how would you cancel it or handle errors inside?